### PR TITLE
Move chart transforms to Grapher

### PR DESCRIPTION
**Pull request by danielgavrilov**
Notion issue: https://www.notion.so/owid/Move-chart-transforms-to-Grapher-d53e023091e245c2914543b7fce0feac

### CODE REVIEW

**Comment by shaahmad** [comment]
Is there a Notion ticket for this? Just to understand on a high-level.

**Comment by shaahmad** [comment]
[Found it](https://www.notion.so/owid/Fully-spec-out-Timeline-when-to-filter-and-all-time-params-728546dbee2146739a94a8c8c7bfb6f6)

**Comment by danielgavrilov** [comment]
Haha, that was quick. I was writing it up. The thing you linked to is a wider project, the issue specifically for this is: https://www.notion.so/owid/Move-chart-transforms-to-Grapher-d53e023091e245c2914543b7fce0feac

File: grapher/core/Grapher.tsx [hunk 1]
```diff
@@ -423,29 +425,47 @@ export class Grapher
...
         table = this.minPopulationFilter
             ? table.filterByPopulation(this.minPopulationFilter)
             : table
-        table = this.filterByTime(table)
+
+        if (
+            this.tab === GrapherTabOption.chart ||
+            this.tab === GrapherTabOption.map
+        ) {
+            const chartComponent = getChartComponent(
```
Comment (breck7) [inline grapher/core/Grapher.tsx:440]: Ah, any idea how to type getChartComponent? I couldn't figure it out earlier. I'll give it another look too.

File: grapher/barCharts/DiscreteBarChart.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { select } from "d3-selection"
 import { min, max, maxBy, sortBy } from "grapher/utils/Util"
 import { computed, action } from "mobx"
 import { observer } from "mobx-react"
+import { bind } from "decko"
```
Comment (breck7) [inline grapher/barCharts/DiscreteBarChart.tsx:6]: If we fix the typing on getChartComponent, can we ditch this?

**Pull request merged**